bootstrap/kubeadm/types/v1beta1/conversion_test.go

```diff
@@ -77,3 +79,11 @@ func clusterConfigurationFuzzer(obj *ClusterConfiguration, c fuzz.Continue) {
 	// ClusterConfiguration.UseHyperKubeImage has been removed in v1alpha4, so setting it to false in order to avoid v1beta1 --> v1alpha4 --> v1beta1 round trip errors.
 	obj.UseHyperKubeImage = false
 }
+
+func kubeadmNodeRegistrationOptionsFuzzer(obj *kubeadmbootstrapv1alpha4.NodeRegistrationOptions, c fuzz.Continue) {
+	c.FuzzNoCustom(obj)
+
+	// NodeRegistrationOptions.IgnorePreflightErrors does not exist in kubeadm v1beta1 API, so setting it to nil in order to avoid
+	// v1alpha4 --> v1beta1 -> v1alpha4 round trip errors.
+	obj.IgnorePreflightErrors = nil
+}
```

Comment:
Shouldn't we restore the field during conversion instead of dropping it?

Reply:
@vincepri 
Very nice finding. I implemented it for the top-level types: KubeadmConfig, KubeadmConfigTemplate and KubeadmControlPlane.

Given that in the current case we convert between InitConfigurations and JoinConfigurations and they don't have ObjectMeta I'm not sure if it's possible in this case.